src/cli/remove.rs

```diff
@@ -41,43 +59,55 @@ pub async fn execute(args: Args) -> miette::Result<()> {
         SpecType::Run
     };
 
-    let results = deps
-        .iter()
-        .map(|dep| {
-            if let Some(p) = &args.platform {
-                project
-                    .manifest
-                    .remove_target_dependency(dep, &spec_type, p)
-            } else {
-                project.manifest.remove_dependency(dep, &spec_type)
-            }
-        })
-        .collect::<Vec<_>>();
-
-    project.save()?;
-
-    // updating prefix after removing from toml
-    let _ = get_up_to_date_prefix(&project, LockFileUsage::Update, false, None).await?;
-
-    for (removed, spec) in results.iter().flatten() {
-        let table_name = if let Some(p) = &args.platform {
-            format!("target.{}.{}", p.as_str(), spec_type.name())
-        } else {
-            spec_type.name().to_string()
-        };
+    let section_name: String = if args.pypi {
+        consts::PYPI_DEPENDENCIES.to_string()
+    } else {
+        spec_type.name().to_string()
+    };
+    let table_name = if let Some(p) = &args.platform {
+        format!("target.{}.{}", p.as_str(), section_name)
+    } else {
+        section_name
+    };
 
+    fn print_ok_dep_removal(pkg_name: &String, pkg_extras: &String, table_name: &String) {
         eprintln!(
             "Removed {} from [{}]",
-            console::style(format!("{removed} {spec}")).bold(),
-            console::style(table_name).bold(),
-        );
+            console::style(format!("{pkg_name} {pkg_extras}")).bold(),
+            console::style(table_name).bold()
+        )
     }
 
-    for result in &results {
-        if let Err(e) = result {
-            eprintln!("{e}");
+    if args.pypi {
+        let all_pkg_name = convert_pkg_name::<rip::types::PackageName>(&deps)?;
+        for dep in all_pkg_name.iter() {
+            let result = if let Some(p) = &args.platform {
+                project.manifest.remove_target_pypi_dependency(dep, p)?
+            } else {
+                project.manifest.remove_pypi_dependency(dep)?
+            };
+            print_ok_dep_removal(
```

Comment:
Printing in this function will possibly give incorrect information to the user as you might error in a next iteration. 

e.g.
```
pixi rm --pypi requests bla
# Results in:
Removed requests "*" from [pypi-dependencies]
  × Couldn't find bla in pypi-dependencies
```
But as the save only happens later it will not actually be removed from the config.

Reply:
indeed, I will change that later today, thanks

Reply:
done!